clang/lib/Tooling/DependencyScanning/IncludeTreeActionController.cpp

```diff
@@ -401,20 +475,40 @@ IncludeTreeBuilder::finishIncludeTree(CompilerInstance &ScanInstance,
       return std::move(E);
   }
 
+  auto &FrontendOpts = NewInvocation.getFrontendOpts();
+  if (!FrontendOpts.Inputs.empty() &&
+      FrontendOpts.Inputs[0].getKind().getFormat() == InputKind::ModuleMap) {
+    // FIXME: handle inferred module maps
+    Expected<FileEntryRef> FE = FM.getFileRef(FrontendOpts.Inputs[0].getFile());
```

Comment:
There was a reason this didn't work when I originally wrote the code, but might not apply anymore. I'll take a look.

Reply:
Oh,  the issue is `addFile` does not return `Expected<IncludeTree:File>`, just `Error`.  I would need to change every call to addFile to change that.  Since this will go away in https://github.com/apple/llvm-project/pull/6519, are you okay to keep it as is?

Reply:
> Since this will go away in https://github.com/apple/llvm-project/pull/6519, are you okay to keep it as is?

SGTM👍